crates/volta-core/src/hook/tool.rs

```diff
@@ -47,6 +52,31 @@ impl DistroHook {
     }
 }
 
+/// Use the expected filename to determine the extension for this hook
+///
+/// This will include the multi-part `tar.gz` extension if it is present, otherwise it will use
+/// the standard extension.
+fn calculate_extension(filename: &str) -> Option<&str> {
+    let mut parts = filename.rsplit('.');
+    match (parts.next(), parts.next(), parts.next()) {
+        (Some(ext), Some("tar"), Some(_)) => {
+            // .tar.gz style extension, return both parts
+            //                          tar  .   gz
+            let index = filename.len() - 3 - 1 - ext.len();
+            filename.get(index..)
+        }
```

Comment:
Could we not just return `"." + tar + "." + ext` (modulo the change to make `+` work here)? It seems like it would be more robust against conceivable variations in extensions, vs. the index-based approach. (Presumably similar would go for the later matches as well.)

Reply:
In this case, it’s the literal value `tar` that we’re looking for, it’s not a generic part, so we can be confident that the index is valid.

We could use something like `”.tar.” + ext`, however that would also require a bit of gymnastics around the return type and creating the new `String`, and I’m not sure if it’s ultimately more readable.

Another option would be to use a regex, though that’s (imo anyway) often even more opaque than any of the existing options. Also a bit of a pain to maintain long-term, because reading regex is always an adventure.

Reply:
I just realized (thanks to @charlespierce telling me so in Discord 😂) that I had misread this; this seems fine, since it's using `ext.len()`—for some reason I totally misread it and thought it was hard coded.